src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -2,18 +2,125 @@
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int INITIAL_CAPACITY = 16;
+    private static final double INITIAL_LOAD_FACTOR = 0.75;
+    private Entry<K, V>[] table = new Entry[INITIAL_CAPACITY];
+    private int size = 0;
+
     @Override
     public void put(K key, V value) {
-
+        if (size >= INITIAL_LOAD_FACTOR * table.length) {
+            resizing();
+        }
+        if (key == null) {
+            putForNullKey(value);
+            return;
+        }
+        int hashCode = key.hashCode();
+        int index = (table.length - 1) & hashCode;
+        Entry<K, V> entry = table[index];
+        if (entry == null) {
+            table[index] = new Entry<>(key, value, null);
```

Comment:
Diamond operator. Be attentive with the diamond operator in java. What is the difference between: `new Node<>(...)` and `new Node<K,V>(...)`and `new Node(...)`. Read more about it [here](https://www.baeldung.com/java-diamond-operator)